oauth2_http/java/com/google/auth/oauth2/IdentityPoolCredentials.java

```diff
@@ -72,17 +80,60 @@ public class IdentityPoolCredentials extends ExternalAccountCredentials {
       throw new IllegalArgumentException(
           "A subjectTokenSupplier or a credentialSource must be provided.");
     }
+
+    // Initialize based on the source type
     if (builder.subjectTokenSupplier != null) {
       this.subjectTokenSupplier = builder.subjectTokenSupplier;
       this.metricsHeaderValue = PROGRAMMATIC_METRICS_HEADER_VALUE;
-    } else if (credentialSource.credentialSourceType
-        == IdentityPoolCredentialSource.IdentityPoolCredentialSourceType.FILE) {
+    } else if (credentialSource.credentialSourceType == IdentityPoolCredentialSourceType.FILE) {
       this.subjectTokenSupplier = new FileIdentityPoolSubjectTokenSupplier(credentialSource);
       this.metricsHeaderValue = FILE_METRICS_HEADER_VALUE;
-    } else {
+    } else if (credentialSource.credentialSourceType == IdentityPoolCredentialSourceType.URL) {
       this.subjectTokenSupplier =
           new UrlIdentityPoolSubjectTokenSupplier(credentialSource, this.transportFactory);
       this.metricsHeaderValue = URL_METRICS_HEADER_VALUE;
+    } else if (credentialSource.credentialSourceType
+        == IdentityPoolCredentialSourceType.CERTIFICATE) {
+      assert credentialSource.certificateConfig != null;
+
+      try {
+        final IdentityPoolCredentialSource.CertificateConfig certConfig =
+            credentialSource.certificateConfig;
+
+        // Use the provided X509Provider if available.
+        X509Provider x509Provider = builder.x509Provider;
+        if (x509Provider == null) {
+          // Determine the certificate path based on the configuration.
+          String explicitCertConfigPath =
+              certConfig.useDefaultCertificateConfig()
+                  ? null
+                  : certConfig.getCertificateConfigLocation();
+
+          // Initialize X509Provider with the explicit path (if provided).
+          x509Provider = new X509Provider(explicitCertConfigPath);
+        }
+
+        // Update the transport factory to use a mTLS transport with the provided certificate.
+        KeyStore mtlsKeyStore = x509Provider.getKeyStore();
+        final NetHttpTransport mtlsTransport =
+            new NetHttpTransport.Builder().trustCertificates(null, mtlsKeyStore, "").build();
+        this.transportFactory = () -> mtlsTransport;
+
+        // Initialize the subject token supplier with the certificate path
+        credentialSource.credentialLocation = x509Provider.getCertificatePath();
```

Comment:
nit: Can we extract this to a helper function so the flow of the constructor generally does:
1. set `subjectTokenSupplier`
2. set `metricsHeaderValue`